packages/plugin-core/src/commands/CopyNoteLink.ts

```diff
@@ -175,13 +186,46 @@ export class CopyNoteLinkCommand extends BasicCommand<
     const editor = VSCodeUtils.getActiveTextEditor()!;
     const fname = NoteUtils.uri2Fname(editor.document.uri);
     const engine = ExtensionProvider.getEngine();
-
     const vault = PickerUtilsV2.getVaultForOpenEditor();
-    const note = NoteUtils.getNoteByFnameFromEngine({
-      fname,
-      vault,
-      engine,
-    }) as NoteProps;
+
+    if (editor.document.isDirty) {
+      this._onEngineNoteStateChangedDisposable =
+        this.engineEvents.onEngineNoteStateChanged(
+          async (noteChangeEntries: NoteChangeEntry[]) => {
+            const savedNote = noteChangeEntries.filter(
+              (entry) => entry.note.fname === fname && entry.status === "update"
+            );
+            // Received event from engine about successful save
+            if (savedNote.length > 0) {
+              await this.executeCopyNoteLink(savedNote[0].note, editor);
+              this.dispose();
```

Comment:
I think we should also use `setTimeout` to dipose of the listener at a certain timeout and also in case anything goes wrong. Otherwise, if something goes wrong in this situation and the user runs the command repeatedly, we may end up with a bunch of zombie subscriptions.

Reply:
Good call